### PR TITLE
[AIR][Serve] Rename ModelWrapperDeployment -> PredictorDeployment

### LINKED ISSUE #25540: [Serve][AIR] Rename ModelWrapperDeployment to PredictorDeployment
Naming was confusing. ModelWrapper naming pre-dates Ray AIR terminologies and it is a AIR predictor deployment afterall.

### PR DESCRIPTION
**Pull request by simon-mo**
Also renamed the file from model_wrappers.py to air_integrations.py

Open to other suggestions

## Related issue number
Closes #25540

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: doc/source/ray-air/examples/rl_serving_example.ipynb [hunk 1]
```diff
@@ -55,7 +55,7 @@
     "from ray.train.rl.rl_trainer import RLTrainer\n",
     "from ray.train.rl.rl_predictor import RLPredictor\n",
     "from ray.air.result import Result\n",
-    "from ray.serve.model_wrappers import ModelWrapperDeployment\n",
+    "from ray.serve.air_integrations import PredictorDeployment\n",
```
Comment (richardliaw) [inline doc/source/ray-air/examples/rl_serving_example.ipynb:58]: Can you do `from ray.serve import PredictorDeployment`?

**Review by richardliaw — CHANGES_REQUESTED** [review]

**Comment by simon-mo** [comment]
Updated. Please take another look

File: doc/source/ray-air/examples/serving_guide.ipynb [hunk 2]
```diff
@@ -6,20 +6,20 @@
...
     "\n",
     "But before that, let's review the key concepts:\n",
     "- [`Checkpoint`](ray.air.checkpoint) represents a trained model stored in memory, file, or remote uri.\n",
     "- [`Predictor`](ray.train.predictor.Predictor)s understand how to perform a model inference given checkpoints and the model definition. Ray AIR comes with predictors for each supported frameworks. \n",
     "- [`Deployment`](serve-key-concepts-deployment) is a Ray Serve construct that represent an HTTP endpoint along with scalable pool of models.\n",
     "\n",
-    "The core concept for deployment is the `ModelWrapper`. The `ModelWrapper` takes a [predictor](ray.train.predictor.Predictor) class and a [checkpoint](ray.air.checkpoint) and transforms them into a live HTTP endpoint. \n",
+    "The core concept for deployment is the `PredictorDeployment`. The `PredictorDeployment` takes a [predictor](ray.train.predictor.Predictor) class and a [checkpoint](ray.air.checkpoint) and transforms them into a live HTTP endpoint. \n",
```
Comment (shrekris-anyscale) [inline doc/source/ray-air/examples/serving_guide.ipynb:20]:
```suggestion
    "The core concept for model deployment is the `PredictorDeployment`. The `PredictorDeployment` takes a [predictor](ray.train.predictor.Predictor) class and a [checkpoint](ray.air.checkpoint) and transforms them into a live HTTP endpoint. \n",
```

File: doc/source/ray-air/examples/serving_guide.ipynb [hunk 3]
```diff
@@ -6,20 +6,20 @@
...
     "\n",
-    "We'll start with a simple quick-start demo showing how you can use the ModelWrapper to deploy your model for online inference."
+    "We'll start with a simple quick-start demo showing how you can use the Predictor to deploy your model for online inference."
```
Comment (shrekris-anyscale) [inline doc/source/ray-air/examples/serving_guide.ipynb:22]:
```suggestion
    "We'll start with a simple quick-start demo showing how you can use the `PredictorDeployment` to deploy your model for online inference."
```

File: doc/source/ray-air/examples/serving_guide.ipynb [hunk 4]
```diff
@@ -282,9 +282,9 @@
    "metadata": {},
    "source": [
     "\n",
-    "It works! As you can see, you can use the `ModelWrapper` to deploy checkpoints trained in Ray AIR as live endpoints. You can find more end-to-end examples for your specific frameworks in the [examples](air-examples-ref) page.\n",
+    "It works! As you can see, you can use the `PredictorDeployment` to deploy checkpoints trained in Ray AIR as live endpoints. You can find more end-to-end examples for your specific frameworks in the [examples](air-examples-ref) page.\n",
     "\n",
-    "This tutorial aims to provide an in-depth understanding of `ModelWrappers`. In particular, it'll demonstrate:\n",
+    "This tutorial aims to provide an in-depth understanding of `PredictorDeployment`. In particular, it'll demonstrate:\n",
```
Comment (shrekris-anyscale) [inline doc/source/ray-air/examples/serving_guide.ipynb:287]:
```suggestion
    "This tutorial aims to provide an in-depth understanding of `PredictorDeployments`. In particular, it'll demonstrate:\n",
```

File: doc/source/ray-air/user-guides.rst [hunk 5]
```diff
@@ -67,7 +67,7 @@ AIR Feature Guides
     +++
     .. link-button:: /ray-air/examples/serving_guide
         :type: ref
-        :text: How to configure AIR ModelWrapper for HTTP input or batching?
+        :text: How to configure AIR Predictor for HTTP input or batching?
```
Comment (shrekris-anyscale) [inline doc/source/ray-air/user-guides.rst:70]:
```suggestion
        :text: How to configure AIR PredictorDeployment for HTTP input or batching?
```

I'm not sure about this– should it be `PredictorDeployment` or `Predictor`?

**Review by shrekris-anyscale — CHANGES_REQUESTED** [review]
Great work! I added some comments.

File: doc/source/ray-air/user-guides.rst (hunk 5 above)
Comment (simon-mo) [reply doc/source/ray-air/user-guides.rst:70]: yeah i think deployment here make sense

**Review by shrekris-anyscale — APPROVED** [review]
Nice work on the updates!

**Review by richardliaw — APPROVED** [review]

**Pull request merged**